pay-api/src/pay_api/models/non_sufficient_funds.py

```diff
@@ -51,6 +51,11 @@ class NonSufficientFundsModel(BaseModel):  # pylint: disable=too-many-instance-a
     invoice_id = db.Column(db.Integer, ForeignKey('invoices.id'), nullable=False)
     invoice_number = db.Column(db.String(50), nullable=False, index=True, comment='CFS Invoice number')
 
+    @classmethod
+    def find_by_invoice_id(cls, invoice_id: int):
```

Comment:
small helper function only works on NonSufficientFunds (no complex joins to other entity types, should be fine)